webapp/commonComponents/expression/expressionEditorPopup.js

```diff
@@ -1,54 +1,96 @@
 import './expressionEditorPopup.scss'
 
-import React from 'react'
+import React, { useState } from 'react'
 import { connect } from 'react-redux'
 import * as R from 'ramda'
 
 import * as Expression from '@core/expressionParser/expression'
 import Popup from '../popup'
 import { useI18n } from '../hooks'
-import ExpressionNode from './nodes/expressionNode'
 
-import { useExpressionEditorPopupState, mapStateToProps } from './expressionEditorPopupState'
+import {
+  useExpressionEditorPopupState,
+  useAdvancedExpressionEditorPopupState,
+  mapStateToProps,
+} from './expressionEditorPopupState'
+
+import AdvancedExpressionEditorPopup from './advancedExpressionEditorPopup'
+import BasicExpressionEditorPopup from './basicExpressionEditorPopup'
 
 const ExpressionEditorPopup = props => {
-  const { nodeDefCurrent, isBoolean, variables, onChange, onClose } = props
+  const { onChange, onClose, hideAdvanced } = props
+
+  const [expressionCanBeApplied, setExpressionCanBeApplied] = useState(false)
 
-  const { query, queryDraft, exprDraft, exprDraftValid, updateDraft } = useExpressionEditorPopupState(props)
+  const {
+    query,
+    queryDraft,
+    exprDraft,
+    exprDraftValid,
+    updateDraft,
+    resetDraft,
+    advanced,
+    setAdvancedEditor,
+  } = useExpressionEditorPopupState({ setExpressionCanBeApplied, ...props })
+
+  const advancedEditorState = useAdvancedExpressionEditorPopupState(props)
```

Comment:
why don't you use useAdvancedExpressionEditorPopupState in the AdvancedExpressionEditorPopup directly?

Reply:
I did this since currently `onApply` juggles between basic/advanced editor state and now that the base component shares the Apply/Reset buttons, it seems quite difficult to do it without access to the state at that level?